src/GitHubExtension/DeveloperId/DeveloperIdProvider.cs

```diff
@@ -102,6 +103,39 @@ public IAsyncOperation<IDeveloperId> LoginNewDeveloperIdAsync()
         }).AsAsyncOperation();
     }
 
+    private DeveloperId LoginNewDeveloperIdWithPAT(Uri hostAddress, SecureString personalAccessToken)
+    {
+        try
+        {
+            GitHubClient gitHubClient = new (new ProductHeaderValue(Constants.DEV_HOME_APPLICATION_NAME), hostAddress);
+            var credentials = new Credentials(new System.Net.NetworkCredential(string.Empty, personalAccessToken).Password);
+            gitHubClient.Credentials = credentials;
+            var newUser = gitHubClient.User.Current().Result;
+            DeveloperId developerId = new (newUser.Login, newUser.Name, newUser.Email, newUser.Url, gitHubClient);
+            SaveOrOverwriteDeveloperId(developerId, personalAccessToken);
+
+            Log.Logger()?.ReportInfo($"{developerId.LoginId} logged in with PAT flow to {developerId.GetHostAddress()}");
+
+            if (developerId is null)
+            {
+                Log.Logger()?.ReportError($"Invalid DeveloperId");
+                throw new InvalidOperationException("Invalid DeveloperId");
+            }
+
+            lock (DeveloperIdsLock)
+            {
+                DeveloperIds.Add(developerId);
+            }

```

Comment:
Quick Question: `SaveOrOverwriteDeveloperId` will also add to the developer id list if there are no duplicates present. Would adding the same developer id to the list again here introduce issues? Is this something required for the GHES flow. Thanks!

Reply:
Oh you're correct. We're adding this to the list twice. Will fix this.

Reply:
Fixed